mutable_buffer/src/chunk.rs

```diff
@@ -152,6 +167,53 @@ impl Chunk {
         }
     }
 
+    pub fn is_open(&self) -> bool {
+        let state = self.state.lock();
+        *state == ChunkState::Open
+    }
+
+    pub fn is_closing(&self) -> bool {
+        let state = self.state.lock();
+        *state == ChunkState::Closing
+    }
+
+    pub fn is_closed(&self) -> bool {
+        let state = self.state.lock();
+        *state == ChunkState::Closed
+    }
+
+    pub fn is_moving(&self) -> bool {
+        let state = self.state.lock();
+        *state == ChunkState::Moving
+    }
+
+    pub fn is_moved(&self) -> bool {
+        let state = self.state.lock();
+        *state == ChunkState::Moved
+    }
+
+    pub fn same_state(&self, chunk_state: ChunkState) -> bool {
+        match chunk_state {
+            ChunkState::Open => self.is_open(),
+            ChunkState::Closing => self.is_closing(),
+            ChunkState::Closed => self.is_closed(),
+            ChunkState::Moving => self.is_moving(),
+            ChunkState::Moved => self.is_moved(),
+        }
+    }
```

Comment:
wouldn't this work (and be more resilient to typos in `is_moving` vs `is_moved` etc)?

```suggestion
    pub fn same_state(&self, chunk_state: ChunkState) -> bool {
        let state = self.state.lock();
        *state == chunk_state
    }
```

Reply:
and perhaps the various `is_moved` et.al. methods could be implemented in terms of `same_state`, like:

```rust
fn is_moving(&self) -> bool { self.same_state(ChunkState::Moving) }
...
```

Reply:
Exactly :)